packages/mobile/src/hooks/useLoadOfflineTracks.ts

```diff
@@ -114,40 +119,105 @@ export const useLoadOfflineTracks = () => {
+  // back online we can refetch fresh content
+  useEffect(() => {
+    if (!isReachable) {
+      setHasGoneOffline(true)
+    }
+  }, [isReachable, setHasGoneOffline])
 
-    const lineupTracks = Object.values(offlineTracks).filter((track) =>
-      track.offline?.reasons_for_download.some(
-        (reason) => reason.collection_id === collectionId
+  // If we go offline, set lineup into a success state with
+  // offline data
+  useEffect(() => {
+    if (isOfflineModeEnabled && collectionId && !isReachable) {
+      const lineupTracks = Object.values(offlineTracks).filter((track) =>
+        track.offline?.reasons_for_download.some(
+          (reason) => reason.collection_id === collectionId.toString()
+        )
       )
-    )
 
-    if (!isReachable) {
-      const populateLinupAction =
-        collectionId === DOWNLOAD_REASON_FAVORITES
-          ? savedPageTracksLineupActions.fetchLineupMetadatasSucceeded(
-              lineupTracks,
-              0,
-              lineupTracks.length,
-              false,
-              false
-            )
-          : collectionPageLineupActions.fetchLineupMetadatasSucceeded(
-              lineupTracks,
-              0,
-              lineupTracks.length,
-              false,
-              false
-            )
-      dispatch(populateLinupAction)
+      if (collectionId === DOWNLOAD_REASON_FAVORITES) {
+        // Reorder lineup tracks accorinding to favorite time
+        const sortedTracks = lineupTracks.sort((a, b) => {
+          const bTime = b.offline?.favorite_created_at
+          const aTime = a.offline?.favorite_created_at
+          if (aTime && bTime) {
+            if (aTime > bTime) {
+              return -1
+            }
+            return 1
+          }
+          return 0
+        })
+        dispatch(
+          lineupActions.fetchLineupMetadatasSucceeded(
+            sortedTracks,
+            0,
+            lineupTracks.length,
+            false,
+            false
+          )
+        )
+      } else {
+        // Reorder lineup tracks according to the collection
+        // TODO: This may have issues for playlists with duplicate tracks
+        const sortedTracks = collection?.playlist_contents.track_ids.map(
+          ({ track: trackId }) =>
+            lineupTracks.find((track) => trackId === track.track_id)
+        )
+        dispatch(
+          lineupActions.fetchLineupMetadatasSucceeded(
+            sortedTracks,
+            0,
+            lineupTracks.length,
+            false,
+            false
+          )
+        )
+      }
+    }
+  }, [
+    dispatch,
+    offlineTracks,
+    isOfflineModeEnabled,
+    lineupActions,
+    isReachable,
+    collectionId,
+    setHasGoneOffline,
+    collection?.playlist_contents.track_ids
+  ])
+
+  // If we go back online, trigger a refetch of the original content
+  useEffect(() => {
+    if (isReachable && hasGoneOffline) {
+      fetchOnlineContent()
```

Comment:
Technically this can be only set once because it's like a "have we done the initial work" check, but I think longer term a nicer model would be to have a single loader hook manage all of this -- everything from first fetch to offline fetches and later refetches.

```
const useLoadLineup = (lineup: 'favorites' | ...) => {
  const doesSupportOffline = lineup === 'favorites' || ...
  if (doesSupportOffline) {
     // reachability stuff
  }

  useFocusEffect() { ... dispatch () }
}

const lineup = useLoadLineup('favorites')

return <Lineup lineup={lineup} />
```

Reply:
(But i made this change anyway)

Reply:
Well I was wondering if you go offline -> online -> offline, do we not need to call the `lineupMetadatasSucceeded` again?

Reply:
the hasGoneOffline is just to make sure that we don't call `fetchOnlineContent` on the first component render, afterwards it should be called every time isReachable becomes true again